### DISCUSSION TITLE
Make Floating Button Available Even When No Note Is Open

### DISCUSSION THREAD

**Discussion by eurghwhy** [chrisgurney/obsidian-note-toolbar#78] [Ideas]
If you install the vConsole plugin, the developer has managed to make a floating button available that is always there regardless of whether a note is open or not. It is also moveable. I think the current floating button could be vastly improved if it used the same approach, particularly on folding phones where the button ends up being positioned in a very weird spot when the phone is unfolded.

**Comment by chrisgurney** [comment]
@mardybardy Interesting suggestion, thank you! Also appreciate the pointer to the [vConsole plugin](https://github.com/zhouhua/obsidian-vconsole)!

All toolbars at the moment are mapped to a note (via a folder, or tag). In this case, it wouldn't be immediately obvious what the toolbar is linked to. Were you envisioning that it would be the global (*) toolbar that would be shown, and otherwise it would be contextual depending on the particular note you're viewing?

As for the issue around foldable phones, I wonder if the [Style Settings plugin](https://github.com/chrisgurney/obsidian-note-toolbar/wiki/Style-Settings-plugin-support) (or CSS snippets) would help here, or not? I'd love to see some screenshots. and the modes your particular phone might be in when you use Obsidian.

**Reply by eurghwhy** [reply]
Yeah I was picturing it on the global toolbar option, or offering the option to have two toolbars if one of them was the button and one of them was something else.

Yeah I believe I would be able to fix it with  some css, for now I think I'm just going to stick to the top menu.

Here are some screenshots of where it positions itself by default on a Galaxy Fold 5 when the screen is unfolded:

Portrait
![Screenshot_20240524_003044_Obsidian](https://github.com/chrisgurney/obsidian-note-toolbar/assets/26926272/d2271d8a-972c-4014-81c3-121d69ea03ae)

Landscape
![Screenshot_20240524_003051_Obsidian](https://github.com/chrisgurney/obsidian-note-toolbar/assets/26926272/636feb00-a5aa-493f-9b20-7fff0935f89e)

**Reply by chrisgurney** [reply]
@mardybardy Thanks for the screenshots! Is the issue that the button for you is too far from the left edge? 

I'm curious what you specifically meant by "the button ends up being positioned in a very weird spot when the phone is unfolded".

**Reply by eurghwhy** [reply]
Yeah it just feels odd compared to where these things normally seem to be. For example here is where it is with the screen folded:

![Screenshot_20240524_003703_Obsidian](https://github.com/chrisgurney/obsidian-note-toolbar/assets/26926272/d7e09d70-5a77-4aa0-859e-a90d236184b8)

That's more natural although I'd still expect it to be further down towards the corner but that might just be me.

**Reply by chrisgurney** [reply]
I can see why it's not further down: It's being placed at 90% of the view height, which means a bigger 10% from the bottom for you, as compared to on a non-folding phone.

I used that approach as it worked the most consistently in my testing.

That said, I would love some eyes on my CSS from some seasoned CSS developers!

The variable at play here is:
`--cg-nt-fab-pos-y: 90vh`

**Comment by chrisgurney** [comment]
See also #200 and #130

**Comment by chrisgurney** [comment]
@mardybardy You are now able to add a toolbar to the _New tab view_ in the [`1.18-beta-02`](https://github.com/chrisgurney/obsidian-note-toolbar/releases/tag/1.18-beta-02) release, if you'd like to try it.

If you [install via BRAT](https://github.com/chrisgurney/obsidian-note-toolbar?tab=readme-ov-file#installation) let me know what you think.

**Comment by chrisgurney** [comment]
Closing now that a toolbar can be set for the _New tab_ view in the latest releases.
